### ISSUE TITLE
PageRemote attribute

### ISSUE THREAD

**Issue by AKTheKnight** [aspnet/AspNetCore.Docs#16920]
Should there not be some documentation on the PageRemote attribute? I can't find any Microsoft documentation on it at all.

I'm using https://www.learnrazorpages.com/razor-pages/validation/remote-validation for now, but I would have expected there to be some information on it here.

Alex

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0156ffc3-47ac-d5da-fbc2-9ff5018b071b
* Version Independent ID: 615db388-a244-41f7-21b5-a4085d9ed479
* Content: [Model validation in ASP.NET Core MVC](https://docs.microsoft.com/en-us/aspnet/core/mvc/models/validation?view=aspnetcore-3.1#feedback)
* Content Source: [aspnetcore/mvc/models/validation.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/mvc/models/validation.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-mvc**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
I agree, however Mike's other article on the same topic is [more accurate](https://www.mikesdotnetting.com/article/343/improved-remote-validation-in-razor-pages). It's not `PageRemoteValidation` as in the article you cite, it's [PageRemoteAttribute](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.pageremoteattribute?view=aspnetcore-3.1)

@serpent5  this look like a fun one. What's I'd do i is create a basic RP (Razor Pages) app and check for a unique `userName`. I'd use in-memory db. Each time you check the DB, if the DB is empty, seed it.

Alternate sample.

Create a RP app with user authorization. You may need to scaffold some of the Identity pages. Add checking for userName or whatever fields Identity requires to be unique.

The Identity sample probably wouldn't be much more complicated and has the advantage of being useful.

**Comment by serpent5** [comment]
Yep, I'll take this. The existing sample is already using RP, so there might be a way to demonstrate this without adding another project. Unless that's definitely what you want...?

**Comment by Rick-Anderson** [comment]
Create a dedicated RP sample app. If you go the auth route, maybe use SQLite so it runs on all platforms.

**Comment by Rick-Anderson** [comment]
@sammychinedu2ky you might like this one. I'm updating your sample so wait until #29202 merges. Or create RP project in lieu of MVC, then you don't need to wait.

**Comment by hishamco** [comment]
@Rick-Anderson I might work on this, can you assign it to me, so I can't forget it accidentally :)
